TABTestKit/Classes/Contexts/NavigationContext.swift

```diff
@@ -13,51 +13,75 @@ public extension NavigationContext {
 	///
 	/// - Parameter screen: The screen to await.
-	func see<ScreenType: Screen>(_ screen: ScreenType) {
-		see(screen.trait)
+	func see<ScreenType: Screen>(_ screen: ScreenType) -> StepAction {
+        return StepAction(description: "see the \(screen.description)") {
+            screen.trait.await(.exists, .visible)
+        }
 	}
 	
 	/// Asserts that a screen does not exist, by awaiting on its trait.
 	///
 	/// - Parameter screen: The screen to await.
-	func doNotSee<ScreenType: Screen>(_ screen: ScreenType) {
-		doNotSee(screen.trait)
+	func doNotSee<ScreenType: Screen>(_ screen: ScreenType) -> StepAction {
+        return StepAction(description: "do not see the \(screen.description)") {
+            screen.trait.await(not: .exists)
+        }
 	}
 	
 	/// Asserts that an element can be seen, by awaiting for it to exist and be visible.
 	///
 	/// - Parameter element: The element to await.
-	func see(_ element: Element) {
-        element.await(.exists, .visible)
+	func see(_ element: Element) -> StepAction {
+        return StepAction(description: "see the \(element.description)") {
+            element.await(.exists, .visible)
+        }
 	}
 	
 	/// Asserts that an element does not exist, by waiting for it to not exist.
 	///
 	/// - Parameter element: The element to await.
-	func doNotSee(_ element: Element) {
-		element.await(not: .exists)
+	func doNotSee(_ element: Element) -> StepAction {
+        return StepAction(description: "do not see the \(element.description)") {
+            element.await(not: .exists)
+        }
 	}
 	
 	/// Completes one or more things that knows how to complete itself.
 	///
 	/// - Parameter completableThings: One or more Completable things. Typically, this would be a Screen that conforms to Completable.
-	func complete(_ completableThings: Completable...) {
+    func complete<CompletableWithDescription: Completable & CustomStringConvertible>(_ completableThings: CompletableWithDescription...) -> StepAction {
+        
 		guard !completableThings.isEmpty else { XCTFatalFail("You must provide at least one Completable thing to complete!") }
-		completableThings.forEach {
-			$0.await()
-			$0.complete()
-		}
+
+        let completableDescription = completableThings
+            .map { $0.description }
+            .joined(separator: " and the ")
+
+        return StepAction(description: "complete the \(completableDescription)") {
+            completableThings.forEach {
+                $0.await()
+                $0.complete()
+            }
+        }
 	}
 	
 	/// Dismisses one or more things that knows how to complete itself.
 	///
 	/// - Parameter dismissableThings: One or more Dismissable things. Typically, this would be a Screen that conforms to Dismissable.
-	func dismiss(_ dismissableThings: Dismissable...) {
+	func dismiss<DismissableWithDescription: Dismissable & CustomStringConvertible>(_ dismissableThings: DismissableWithDescription...) -> StepAction {
```

Comment:
Not sure we need to break everyone's implementation to add some descriptions, I would expect descriptions to be optional since there are probably some pretty big codebases out there using TABTestKit and expecting everyone to have to go through and add descriptions to everything might be really tedious. Otherwise they might get stuck on an old version, not having time to update everything, and then will miss out on small improvements and additional elements etc

Reply:
This PR has the effect that Elements & Screens will be CustomStringConvertible by default, yes you can override it but there is a default value.
(please correct me if I'm wrong) but in general only Elements & Screens will be Dismissable, therefore no change is needed on the clients side.
That being said I do think this change will lead to a major version bump.

Reply:
Ahhhh okay that makes more sense